### ISSUE TITLE
BBC: Blead Breaks PPR

### ISSUE THREAD

**Issue by cjg-cguevara** [Perl/perl5#20847]
This is a bug report for perl from "Carlos Guevara" <carlos@carlosguevara.com>,
generated with the help of perlbug 1.43 running under perl 5.37.10.

-----------------------------------------------------------------

BBC: Blead Breaks PPR

Please see http://fast-matrix.cpantesters.org/?dist=PPR

---
**Flags**
- category=core
- severity=low
---
**Perl configuration**
```
Site configuration information for perl 5.37.10:

Configured by cpan at Tue Feb 21 09:34:13 EST 2023.

Summary of my perl5 (revision 5 version 37 subversion 10) configuration:
  Commit id: dab4006c92d70a54e7909d0e670b71bfd3fb8221
  Platform:
    osname=linux
    osvers=5.15.94-0-lts
    archname=x86_64-linux
    uname='linux cjg-alpine3 5.15.94-0-lts #1-alpine smp thu, 16 feb 2023 21:49:19 +0000 x86_64 linux '
    config_args='-des -Dprefix=~/bin/perl-blead -Dscriptdir=~/bin/perl-blead/bin -Dusedevel -Duse64bitall'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-O2'
    cppflags='-D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong'
    ccversion=''
    gccversion='12.2.1 20220924'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib /lib
    libs=-lpthread -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -ldl -lm -lcrypt -lutil -lc
    libc=/usr/lib/libc.a
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'


---
@INC for perl 5.37.10:
    /home/cpan/bin/perl-blead/lib/site_perl/5.37.10/x86_64-linux
    /home/cpan/bin/perl-blead/lib/site_perl/5.37.10
    /home/cpan/bin/perl-blead/lib/5.37.10/x86_64-linux
    /home/cpan/bin/perl-blead/lib/5.37.10

---
Environment for perl 5.37.10:
    HOME=/home/cpan
    LANG=C.UTF-8
    LANGUAGE (unset)
    LC_COLLATE=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/cpan/bin/perl-blead/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash
```

**Comment by demerphq** [comment]
Patch filed as part of https://rt.cpan.org/Ticket/Display.html?id=146681

**Comment by demerphq** [comment]
PPR patch has been merged, and version 0.001007 has been released. Builds and installs on latest perl just fine. 

```
$ cpanm PPR
--> Working on PPR
Fetching http://www.cpan.org/authors/id/D/DC/DCONWAY/PPR-0.001007.tar.gz ... OK
Configuring PPR-0.001007 ... OK
Building and testing PPR-0.001007 ... OK
Successfully installed PPR-0.001007
1 distribution installed
$ perl -v
This is perl 5, version 37, subversion 10 (v5.37.10 (v5.37.9-32-g10c7b2021d)) built for x86_64-linux-thread-multi
```
